[HttpStress] [SslStress] Linux stress test builds are broken

The `Build CLR And Libraries` phase is failing starting with April 13:

https://dev.azure.com/dnceng-public/public/_build/results?buildId=251483&view=logs&j=22fea640-1099-5f32-ec5d-316ad83f359a&t=ebe1360c-fe50-5ebc-628a-48ef9bf66506

```
2023-04-13T13:23:26.6081570Z   ILCompiler -> /repo/artifacts/bin/coreclr/linux.x64.Release/ilc/ilc.dll
2023-04-13T13:23:26.7889770Z   Generating native code
2023-04-13T13:23:50.3607628Z clang-9 : error : invalid linker name in argument '-fuse-ld=lld' [/repo/src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj]
2023-04-13T13:23:50.3747182Z ##[error]clang-9(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) invalid linker name in argument '-fuse-ld=lld'
2023-04-13T13:23:50.3757194Z /repo/.packages/microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/build/Microsoft.NETCore.Native.targets(350,5): error MSB3073: The command ""/usr/local/bin/clang-9" "/repo/artifacts/obj/coreclr/ILCompiler/x64/Release/native/ilc.o" -o "/repo/artifacts/bin/coreclr/linux.x64.Release/ilc/native/ilc" -fuse-ld=lld /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libbootstrapper.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libRuntime.ServerGC.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libstdc++compat.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libnumasupportdynamic.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Globalization.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.IO.Compression.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Net.Security.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Security.Cryptography.Native.OpenSsl.a -g -Wl,-rpath,'$ORIGIN' -Wl,--build-id=sha1 -Wl,--as-needed -pthread -ldl -lz -lrt -lm -pie -Wl,-z,relro -Wl,-z,now -Wl,--discard-all" exited with code 1. [/repo/src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj]
2023-04-13T13:23:50.3761088Z ##[error]/repo/.packages/microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/build/Microsoft.NETCore.Native.targets(350,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command ""/usr/local/bin/clang-9" "/repo/artifacts/obj/coreclr/ILCompiler/x64/Release/native/ilc.o" -o "/repo/artifacts/bin/coreclr/linux.x64.Release/ilc/native/ilc" -fuse-ld=lld /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libbootstrapper.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libRuntime.ServerGC.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libstdc++compat.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/sdk/libnumasupportdynamic.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Globalization.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.IO.Compression.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Net.Security.Native.a /repo/.packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/8.0.0-preview.1.23110.8/framework/libSystem.Security.Cryptography.Native.OpenSsl.a -g -Wl,-rpath,'$ORIGIN' -Wl,--build-id=sha1 -Wl,--as-needed -pthread -ldl -lz -lrt -lm -pie -Wl,-z,relro -Wl,-z,now -Wl,--discard-all" exited with code 1.
2023-04-13T13:23:50.3830354Z 
2023-04-13T13:23:50.3831558Z Build FAILED.
```

The most likely cause is that the container has different versions of clang and lld installed, the issue is explained in https://github.com/dotnet/runtime/issues/84794#issuecomment-1509601271.

--- a/eng/docker/libraries-sdk.linux.Dockerfile
+++ b/eng/docker/libraries-sdk.linux.Dockerfile
@@ -1,8 +1,29 @@
 # Builds and copies library artifacts into target dotnet sdk image
-ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-f39df28-20191023143754
+ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04
 ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/nightly/sdk:7.0-bullseye-slim
 
 FROM $BUILD_BASE_IMAGE as corefxbuild
+
+# Enable openssl legacy provider in system openssl config
+RUN fixOpensslConf=$(mktemp) && \
+    printf "#!/usr/bin/env sh\n\
+        sed -i '\n\
+            # Append 'legacy = legacy_sect' after 'default = default_sect' under [provider_sect]
+            /^default = default_sect/a legacy = legacy_sect\n\
+            # Search for [default_sect]
+            /\[default_sect\]/ {\n\
+                # Go to next line
+                n\n\
+                # Uncomment '# activate = 1'
+                s/# //\n\
+                # Append new [legacy_sect], with 'activate = 1'
+                a\n\
+                a [legacy_sect]\n\
+                a activate = 1\n\
+            }\n\
+            ' /etc/ssl/openssl.cnf\n" > $fixOpensslConf && \
+    sh $fixOpensslConf && \
+    rm $fixOpensslConf
 
 ARG CONFIGURATION=Release
 

--- a/eng/docker/libraries-sdk.linux.Dockerfile
+++ b/eng/docker/libraries-sdk.linux.Dockerfile
@@ -1,29 +1,8 @@
 # Builds and copies library artifacts into target dotnet sdk image
-ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7
+ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8
 ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/nightly/sdk:7.0-bullseye-slim
 
 FROM $BUILD_BASE_IMAGE as corefxbuild
-
-# Enable openssl legacy provider in system openssl config
-RUN fixOpensslConf=$(mktemp) && \
-    printf "#!/usr/bin/env sh\n\
-        sed -i '\n\
-            # Append 'legacy = legacy_sect' after 'default = default_sect' under [provider_sect]
-            /^default = default_sect/a legacy = legacy_sect\n\
-            # Search for [default_sect]
-            /\[default_sect\]/ {\n\
-                # Go to next line
-                n\n\
-                # Uncomment '# activate = 1'
-                s/# //\n\
-                # Append new [legacy_sect], with 'activate = 1'
-                a\n\
-                a [legacy_sect]\n\
-                a activate = 1\n\
-            }\n\
-            ' /etc/ssl/openssl.cnf\n" > $fixOpensslConf && \
-    sh $fixOpensslConf && \
-    rm $fixOpensslConf
 
 ARG CONFIGURATION=Release
 

--- a/eng/docker/libraries-sdk.linux.Dockerfile
+++ b/eng/docker/libraries-sdk.linux.Dockerfile
@@ -1,5 +1,5 @@
 # Builds and copies library artifacts into target dotnet sdk image
-ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04
+ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7
 ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/nightly/sdk:7.0-bullseye-slim
 
 FROM $BUILD_BASE_IMAGE as corefxbuild

